cli.py

```diff
@@ -0,0 +1,72 @@
+import os
+from argparse import ArgumentParser
+
+from processor import OsdGenConfig, OsdGenerator
+
+if __name__ == '__main__':
+
+    parser = ArgumentParser()
+    parser.add_argument('--video-path', help='Path to the video file',
+                        required=True)
+    parser.add_argument('--osd-path', help='Path to the OSD file. If none '
+                                           'specified, it will look in the same'
+                                           ' directory as the video path')
+    parser.add_argument('--srt-path', help='Path to SRT file. If none '
+                                           'specified, it will look in the same'
+                                           ' directory as the video path')
+    parser.add_argument('--font-path', required=True,
+                        help='Path to font file - e.g (INAV_36.png)')
+    parser.add_argument('--offset-top', type=int, default=0,
+                        help='Offset from top for OSD')
+    parser.add_argument('--offset-left', help='Offset from left for OSD',
+                        default=0, type=int)
+    parser.add_argument('--osd-zoom', type=int, default=100,
+                        help='Scaling of OSD elements')
+    parser.add_argument('--render-upscale', default=False,
+                        help='Increase video dimensions to 1440p')
+    parser.add_argument('--include-srt', action='store_true', default=False,
+                        help='Include SRT data from VTX')
+    parser.add_argument('--hide-sensitive-osd', action='store_true',
+                        help='Hide Sensitive Elements like lat, lon, altitude, '
+                             'home', default=False)
+    parser.add_argument('--use-hw', action='store_true', default=False,
+                        help='Use Hardware Acceleration for Encoding')
+    parser.add_argument('--fast-srt', action='store_true', default=False,
+                        help='')
+
+    args = parser.parse_args()
+
+    def implicit_path(args, ext):
+        """
+        Attempts to find a path for the OSD and SRT files implicitly
+        file_path: path to the video (mp4) file
+        ext: file type suffix (e.g. osd or srt)
+        """
+        if ext not in {'osd', 'srt'}:
+            raise ValueError('Invalid file type, only srt or osd')
+        if getattr(args, f'{ext}_path') is None:
+            folder, _ = os.path.splitext(args.video_path)
+            return f"{folder}.{ext}"
+        else:
+            return args[f'{ext}_path']
+
+
+    generator_config = OsdGenConfig(
+        video_path=args.video_path,
+        osd_path=implicit_path(args, 'osd'),
+        srt_path=implicit_path(args, 'srt'),
+        font_path=args.font_path,
+        output_path='/tmp/files_out3', # TODO make this video only and png automatic
```

Comment:
Let's make this an input parameter with default value of current directory

Reply:
yep good call for now - just I guess eventually it would be good to automatically clean up the PNGs after rendering. Is there a use case for keeping them?

Reply:
I'm using only PNG files actually! I never render osd directly on video 🤷 
So there should be a switch maybe just to render png? and stop. Then another option to render video and remove png. That makes a lot of sense to me.

Reply:
maybe just adding a 

--cleanup-pngs (default=False) help='removes PNGs after rendering video'. This would only work if you're actually rendering video.

See below about what the API looks like but yes you could also add a --render-video flag if you wanted ffmpeg to do the overlay on the video. As you can see with what I've done I've implicitly assumed everyone just wants ffmpeg to do it!